addons/website/tests/test_redirect.py

```diff
@@ -67,3 +67,170 @@ def test_sitemap_with_redirect(self):
             urls = [url['loc'] for url in pages]
             self.assertIn('/website/info', urls)
             self.assertNotIn('/test', urls)
+
+
+@tagged('-at_install', 'post_install')
+class TestConditionalRedirects(HttpCase):
+
+    def setUp(self):
+        super().setUp()
+        self.test_group = self.env['res.groups'].create({'name': 'Test Group'})
+        # User who is a member of the test group
+        self.user_in_group = self.env['res.users'].with_context({'no_reset_password': True}).create({
+            'name': 'Test User In Group',
+            'login': 'test_user_in_group',
+            'password': 'test_user_in_group',
+            'email': 'test_user_in_group@mail.com',
+            'group_ids': [(6, 0, [self.env.ref('base.group_portal').id, self.test_group.id])]
+        })
+        # User who is NOT a member of the test group
+        self.user_not_in_group = self.env['res.users'].with_context({'no_reset_password': True}).create({
+            'name': 'Test User Not In Group',
+            'login': 'test_user_not_in_group',
+            'password': 'test_user_not_in_group',
+            'email': 'test_user_not_in_group@mail.com',
+            'group_ids': [(6, 0, [self.env.ref('base.group_portal').id])]
+        })
+        self.original_page = self.env['website.page'].create({
+            'name': 'Original Page',
+            'url': '/original-page',
+            'is_published': False,
+            'type': 'qweb',
+            'arch': '<div>Original Page Content</div>',
+        })
+        self.redirected_page = self.env['website.page'].create({
+            'name': 'Redirected Page',
+            'url': '/redirected-page',
+            'is_published': True,
+            'type': 'qweb',
+            'arch': '<div>Redirected Page Content</div>',
+        })
+        self.url_from_301 = '/original-page?redirect=301'
+        self.url_to_301 = '/redirected-page?redirect=301'
+        self.url_from_302 = '/original-page?redirect=302'
+        self.url_to_302 = '/redirected-page?redirect=302'
+        self.url_from_404 = '/website/info'
```

Comment:
You can simplify this, and later its usage can minimize the test lines as well. 😉 

```suggestion
        # URL constants
        self.urls = {
            '301': {'from': '/original-page?redirect=301', 'to': '/redirected-page?redirect=301'},
            '302': {'from': '/original-page?redirect=302', 'to': '/redirected-page?redirect=302'},
            '404': {'from': '/website/info', 'to': None},
        }
```